src/txn/txn.c

```diff
@@ -381,7 +396,8 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 	 */
 	__wt_txn_release_snapshot(session);
 	txn->isolation = session->isolation;
-	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID | WT_TXN_RUNNING);
+	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID |
+	    WT_TXN_NAMED_SNAPSHOT | WT_TXN_READONLY | WT_TXN_RUNNING);
```

Comment:
I couldn't find a path where `WT_TXN_AUTOCOMMIT` was important to maintain across release, but I could be wrong?

(As far as I can tell, WT_TXN_AUTOCOMMIT is set by the appropriate API calls, and, in fact, I would think it must be set by those calls, otherwise, the first use of the allocated object wouldn't work?)